source/gui/addonGui.py

```diff
@@ -489,8 +489,7 @@ def installAddon(parentWindow: wx.Window, addonPath: str) -> bool:  # noqa: C901
 		from gui._addonStoreGui.controls.messageDialogs import _shouldInstallWhenAddonTooOldDialog
 		if _shouldInstallWhenAddonTooOldDialog(parentWindow, bundle._addonGuiModel):
 			# Install incompatible version
-			if not bundle.overrideIncompatibility:

```

Comment:
Are you sure this check is no longer needed?

This covers the case of: an installed add-on, currently with overriden compatibility, updating to another incompatible version. I imagine in this case the pending state will be added even though the add-on is already marked as compatible

Reply:
> This covers the case of: an installed add-on, currently with overriden compatibility, updating to another incompatible version. I imagine in this case the pending state will be added even though the add-on is already marked as compatible

Yes, and that is exactly what should happen. Consider the following:
User has installed version 1.0 of incompatible add-on overriding compatibility, so add-on is now in the overridden compat state. Then they upgraded to the version 1.1, which is still incompatible. With the code in current master the check I have removed meant that the add-on won't be added to the overridden compat state, but when removing version 1.0 before version 1.1 gets installed, the user choice of overridden compat would be lost. Now the add-on is unconditionally added to the pending override compatibility state as long as user decided to override its compatibility, so it can be safely removed from the  overridden compat when uninstalling version 1.0, as it would be added there again when installing version 1.1.

Reply:
Thanks for explaining